### PR TITLE
buildbot/master.cfg: Use CMake on Windows

### PR DESCRIPTION
**Pull request by spycrab**
Untested at the moment, should work if Ninja, CMake and CL are in PATH.

### CODE REVIEW

File: buildbot/master.cfg [hunk 1]
```diff
@@ -126,38 +126,38 @@ def make_dolphin_win_build(build_type, arch, mode="normal"):
...
 
     branch = WithProperties("%s", "branchname")
     env = {"DOLPHIN_BRANCH": branch, "DOLPHIN_DISTRIBUTOR": "dolphin-emu.org"}
     if normal:
         env["DOLPHIN_DEFAULT_UPDATE_TRACK"] = "beta"
-    f.addStep(Compile(command=["msbuild.exe", "/v:m", "/p:Platform=%s" % msarch,
-                               "/p:Configuration=%s" % build_type,
-                               "dolphin-emu.sln"],
+    f.addStep(Compile(command=["cmake", "-GNinja", "-DCMAKE_BUILD_TYPE=%s" % (build_type), ".."],
+                      env=env, workdir="configuring",
```
Comment (BhaaLseN) [inline buildbot/master.cfg]: Not sure if you meant `workdir="configuring"` or `description="configuring"`. Seems odd to have a directory name ending in -ing.

**Comment by delroth** [comment]
- This does not do incremental builds. This is a waste of CPU time, and a regression compared to the current situation. All other platforms are doing incremental builds right now.
- There is no documentation as to how to build Dolphin without VC++ right now.
- The CMake migration does not pin to a given WinSDK version like the previous MSVC projects were doing, and it does not auto-identify path to the compiler from the WinSDK (which means for every WinSDK upgrade we have to go and update our PATH manually and all). This does not seem acceptable to me.

File: buildbot/master.cfg [hunk 2]
```diff
@@ -126,38 +126,40 @@ def make_dolphin_win_build(build_type, arch, mode="normal"):
...
-                      workdir="build/Source",
+                      workdir="build/build",
                       description="building",
                       descriptionDone="build",
                       haltOnFailure=True))
-    f.addStep(Test(command=["msbuild.exe", "/v:m", "/p:Platform=%s" % msarch,
-                               "/p:Configuration=%s" % build_type,
-                               "/p:RunUnitTests=true",
-                               "dolphin-emu.sln"],
+    f.addStep(Test(command=["ninja unittests"],
```
Comment (degasus) [inline buildbot/master.cfg]: I assume ["ninja", "unittests"] instead?

**Comment by degasus** [comment]
> * The CMake migration does not pin to a given WinSDK version like the previous MSVC projects were doing, and it does not auto-identify path to the compiler from the WinSDK (which means for every WinSDK upgrade we have to go and update our PATH manually and all). This does not seem acceptable to me.

This is sadly common on windows. They usually have some magic shell with all environment variables for their compiler. Also one for x64 and one for x86...

**Comment by spycrab** [comment]
> There is no documentation as to how to build Dolphin without VC++ right now.

I mean there's this:
https://github.com/dolphin-emu/dolphin/pull/8076/files#diff-1e290ac8433d555bce009b162cb869d0

Or is this not what you mean?

**Comment by altimumdelta** [comment]
> > * The CMake migration does not pin to a given WinSDK version like the previous MSVC projects were doing, and it does not auto-identify path to the compiler from the WinSDK (which means for every WinSDK upgrade we have to go and update our PATH manually and all). This does not seem acceptable to me.
> 
> This is sadly common on windows. They usually have some magic shell with all environment variables for their compiler. Also one for x64 and one for x86...

Yeah, I came across similar stuff when building FFmpeg on Windows with MSYS/MINGW for the framedumping upgrade attempts, there's a step when Developer Command Prompt for Visual Studio x64 is required (tho weirdly in my case the VS2017 install did not create a shortcut for the x64 version AFAIK which is made by using -arch64 argument), and it uses some kind of pre-initialization environment which pulls in env variables from various places. 

`C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\Common7\Tools\VsDevCmd.bat`

And there's another thing Native Tools Command Prompt, I forgot a bit how they're related or not but I think I had to use only one for my case.

`C:\Program Files (x86)\Microsoft Visual Studio\2017\Community\VC\Auxiliary\Build\vcvars64.bat`

Anyway the point is, maybe something inside there could help in some fashion.

**Comment by stenzek** [comment]
`vcvars64.bat` calls `VsDevCmd.bat`. We want the x64 native compiler (produces 64 bit code and the compiler is 64-bit itself), so `"C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Auxiliary\Build\vcvars64.bat"` should be the one we're running, or `VsDevCmd.bat x64`. This should also set the Windows SDK path.

**Comment by delroth** [comment]
@spycrab I was expecting something at least as well documented as https://github.com/dolphin-emu/dolphin/wiki/Building-for-Windows

Re: vcvars, yes, I know about vcvars, thank you. That does not answer my question regarding Windows SDK / toolchain pinning (which we have needed in the past multiple times, e.g. because of newer Windows SDK dropping platforms support, or bugs in newer versions that required holdback). Our Buildbot has also never used vcvars before and this PR has no changes related to this.

**Comment by Warepire** [comment]
Could something like this help the incremental build issue if CMake doesn't provide it's own  way of enabling that: https://github.com/frerich/clcache

**Comment by spycrab** [comment]
@delroth I wasn't going to touch that until we've actually switched to CMake. Doing that preemptively just seems silly to me.

**Pull request closed without merging**